src/transaction_builder.js

```diff
@@ -453,7 +486,7 @@ TransactionBuilder.prototype.sign = function (index, keyPair, redeemScript, hash
   }
 
   // ready to sign?
-  var signatureScript = input.redeemScript || input.prevOutScript
+  signatureScript = signatureScript || input.redeemScript || input.prevOutScript
```

Comment:
preperation for segwit, where the `signatureScript` could be different from the `redeemScript`